collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.kt

```diff
@@ -0,0 +1,109 @@
+import org.odk.collect.android.projects.CurrentProjectProvider
+import org.odk.collect.android.utilities.FormsRepositoryProvider
+import org.odk.collect.android.utilities.InstancesRepositoryProvider
+import org.odk.collect.geo.SelectionMapFragment
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.strings.localization.LocalizedActivity
+import javax.inject.Inject
+
+/**
+ * Show a map with points representing saved instances of the selected form.
+ */
+class FormMapActivity : LocalizedActivity() {
+
+    @Inject
+    lateinit var formsRepositoryProvider: FormsRepositoryProvider
+
+    @Inject
+    lateinit var instancesRepositoryProvider: InstancesRepositoryProvider
+
+    @Inject
+    lateinit var settingsProvider: SettingsProvider
+
+    @Inject
+    lateinit var currentProjectProvider: CurrentProjectProvider
+
+    private lateinit var viewModel: FormMapViewModel
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        DaggerUtils.getComponent(this).inject(this)
+        setContentView(R.layout.form_map_activity)
+
+        val formId = intent.getLongExtra(EXTRA_FORM_ID, -1)
+        viewModel = ViewModelProvider(
+            this,
+            object : ViewModelProvider.Factory {
+                override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                    return FormMapViewModel(
+                        resources,
+                        formId,
+                        formsRepositoryProvider.get(),
+                        instancesRepositoryProvider.get(),
+                        settingsProvider
+                    ) as T
+                }
+            }
+        )[ARG_VIEW_MODEL_KEY, FormMapViewModel::class.java]
+
+        val formNavigator = FormNavigator(
+            currentProjectProvider,
+            settingsProvider,
+            instancesRepositoryProvider::get
+        )
+
+        supportFragmentManager.setFragmentResultListener(
+            SelectionMapFragment.REQUEST_SELECT_ITEM,
+            this
+        ) { _: String?, result: Bundle ->
+            if (result.containsKey(SelectionMapFragment.RESULT_SELECTED_ITEM)) {
+                val instanceId = result.getLong(SelectionMapFragment.RESULT_SELECTED_ITEM)
+                formNavigator.editInstance(this, instanceId)
+            } else if (result.containsKey(SelectionMapFragment.RESULT_CREATE_NEW_ITEM)) {
+                formNavigator.newInstance(this, formId)
+            }
+        }
+
+        supportFragmentManager.beginTransaction()
+            .add(
+                R.id.selection_map,
+                SelectionMapFragment::class.java,
+                Bundle().also {
+                    it.putString(SelectionMapFragment.ARG_VIEW_MODEL_KEY, ARG_VIEW_MODEL_KEY)
+                }
+            )
+            .commit()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.load()
```

Comment:
We probably could use the result API. Reloading data in `onResume` is an existing part of the Form Map functionality that we could change, but I think we should prioritize moving on to the GeoJSON work rather than going down that avenue here. What do you think?

Reply:
Ok we can skip it for now but please keep in mind that this is not the best solution so that maybe when you have time later you can fix it too.